File: CS/DisplayGroupSummaries_MVVM/MainViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Mvvm.Xpf;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;

namespace DisplayGroupSummaries_MVVM {
    public class Account {
        public string UserName { get; set; }
        public DateTime RegistrationDate { get; set; }
        public bool Married { get; set; }
        public int Age { get; set; }
    }
    public class MainViewModel : ViewModelBase {
        public ObservableCollection<Account> AccountList{ get; }
        public MainViewModel() {
            AccountList = new ObservableCollection<Account>(GetAccountList());
        }
        IEnumerable<Account> GetAccountList() {
            yield return new Account() {
                UserName = "Nick White",
                RegistrationDate = DateTime.Today,
                Married = true,
                Age = 42
            };
            yield return new Account() {
                UserName = "Jack Rodman",
                RegistrationDate = new DateTime(2009, 5, 10),
                Married = false,
                Age = 30
            };
            yield return new Account() {
                UserName = "Sandra Sherry",
                RegistrationDate = new DateTime(2008, 12, 22),
                Married = false,
                Age = 18
            };
            yield return new Account() {
                UserName = "Sabrina Vilk",
                RegistrationDate = DateTime.Today,
                Married = true,
                Age = 24
            };
            yield return new Account() {
                UserName = "Mike Pearson",
                RegistrationDate = new DateTime(2008, 10, 18),
                Married = true,
                Age = 37
            };
        }

        [Command]
        public void CustomSummaryExistsCommand(RowSummaryExistsArgs args) {
            args.Exists = args.GroupPath[0].GroupLevel == 0;
        }
    }
}

File: VB/DisplayGroupSummaries_MVVM/MainViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Mvvm.Xpf
Imports System
Imports System.Collections.Generic
Imports System.Collections.ObjectModel

Namespace DisplayGroupSummaries_MVVM

    Public Class Account

        Public Property UserName As String

        Public Property RegistrationDate As Date

        Public Property Married As Boolean

        Public Property Age As Integer
    End Class

    Public Class MainViewModel
        Inherits ViewModelBase

        Public ReadOnly Property AccountList As ObservableCollection(Of Account)

        Public Sub New()
            AccountList = New ObservableCollection(Of Account)(GetAccountList())
        End Sub

        Private Iterator Function GetAccountList() As IEnumerable(Of Account)
            Yield New Account() With {.UserName = "Nick White", .RegistrationDate = Date.Today, .Married = True, .Age = 42}
            Yield New Account() With {.UserName = "Jack Rodman", .RegistrationDate = New DateTime(2009, 5, 10), .Married = False, .Age = 30}
            Yield New Account() With {.UserName = "Sandra Sherry", .RegistrationDate = New DateTime(2008, 12, 22), .Married = False, .Age = 18}
            Yield New Account() With {.UserName = "Sabrina Vilk", .RegistrationDate = Date.Today, .Married = True, .Age = 24}
            Yield New Account() With {.UserName = "Mike Pearson", .RegistrationDate = New DateTime(2008, 10, 18), .Married = True, .Age = 37}
        End Function

        <Command>
        Public Sub CustomSummaryExistsCommand(ByVal args As RowSummaryExistsArgs)
            args.Exists = args.GroupPath(0).GroupLevel = 0
        End Sub
    End Class
End Namespace
